### PR TITLE
fix unauthenticated page request not remembered post authentication

### LINKED ISSUE #2066: Unauthenticated page request not remembered post authentication
When trying to request a protected page (e.g. http://localhost:8080/#/user-management) while still unauthenticated, the user should be shown the login screen and then be automatically brought to the originally requested page. This isn't currently working. Instead, the user is directed to the home page post-login.

Pre
- unauthenticated

Steps
- request http://localhost:8080/#/user-management
- submit admin credentials

Expected
- continue to http://localhost:8080/#/user-management (authenticated)

Actual
- continue to home (i.e. http://localhost:8080/#/) (authenticated)

### PR DESCRIPTION
**Pull request by sdoxsee**
remove unused logout state/controller. amalgamate previousState and returnToState as they appear to have the same purpose. reset previousState in Auth.logout(). stop recording new previousState after successful state change if we're going to login or if the previousState has just been reset.

fix #2066

### CODE REVIEW

**Comment by andidev** [comment]
I'll have a look into this. Note that you also need to remove the removed files programmatically in https://github.com/jhipster/generator-jhipster/blob/master/app/index.js#L1124

File: app/templates/src/main/webapp/scripts/app/_app.js [hunk 1]
```diff
@@ -25,8 +25,14 @@ angular.module('<%=angularAppName%>', ['LocalStorageModule', <% if (enableTransl
...
             var titleKey =<% if (enableTranslation) { %> 'global.title' <% }else { %> '<%= baseName %>' <% } %>;
 
-            $rootScope.previousStateName = fromState.name;
-            $rootScope.previousStateParams = fromParams;
+            // Remember previous state unless we've been redirected to login or we've just
+            // reset the state memory after logout. If we're redirected to login, our
+            // previousState is already set in the authExpiredInterceptor. If we're going
+            // to login directly, we don't want to be sent to some previous state anyway
+            if (toState.name != 'login' && $rootScope.previousStateName) {
+              $rootScope.previousStateName = fromState.name;
```
Comment (andidev) [inline app/templates/src/main/webapp/scripts/app/_app.js]: Wrong indentation. It should be 4 spaces

**Comment by andidev** [comment]
So the logout state where never used? 
Not sure about this but maybe there is a point in keeping it so that the user can logout manually by the url if wanted.

**Comment by sdoxsee** [comment]
I didn't know about removing the files programatically as well. Thanks for catching that!

I don't think the logout state is ever used unless you type the URL into the browser. I would consider that dead code. If someone wanted to add their own logout URL state again later then they could always do so. Seems that jhipster should avoid generating unused code.

**Comment by andidev** [comment]
I guess you are right.

@jdubois what do you think? 
How come we hava a logout controller in angular which is never used? Is it okay to remove?

**Comment by jdubois** [comment]
Yes I coded it at the beginning, not sure this is being used!!
Let me test this and merge!

**Pull request merged**